pkg/controller/kubelet-config/helpers.go

```diff
@@ -165,15 +169,6 @@ func updateOriginalKubeConfigwithNodeConfig(node *osev1.Node, originalKubeletCon
 
 // updateMachineConfigwithCgroup updates the Machine Config object based on the cgroup mode present in the Config Node resource.
 func updateMachineConfigwithCgroup(node *osev1.Node, mc *mcfgv1.MachineConfig) error {
-	if node == nil {
```

Comment:
If we remove this check, could we panic below for node.spec? i.e. is there no possible scenario where this is nil?

Reply:
Correct, there is no possibility of a crash here. The API structure does not contain pointer values.

https://github.com/openshift/api/blob/c90b4c82852ead1fdb58866b685fe75b0b62163d/config/v1/types_node.go#L19-L45